junit-jupiter-api/src/main/kotlin/org/junit/jupiter/api/Assertions.kt

```diff
@@ -121,3 +121,42 @@ inline fun <reified T : Throwable> assertThrows(message: String, noinline execut
  */
 inline fun <reified T : Throwable> assertThrows(noinline message: () -> String, noinline executable: () -> Unit): T =
     Assertions.assertThrows(T::class.java, Executable(executable), Supplier(message))
+
+/**
+ * Example usage:
+ * ```kotlin
+ * assertDoesNotThrow {
+ *     // Code block that is expected to not throw an exception
+ * }
+ * ```
+ * @see Assertions.assertDoesNotThrow
+ */
+@API(status = EXPERIMENTAL, since = "5.5")
+fun assertDoesNotThrow(executable: () -> Unit): Unit =
+    Assertions.assertDoesNotThrow(Executable(executable))
+
+/**
+ * Example usage:
+ * ```kotlin
+ * assertDoesNotThrow("Should not throw an exception") {
+ *     // Code block that is expected to not throw an exception
+ * }
+ * ```
+ * @see Assertions.assertDoesNotThrow
+ */
+@API(status = EXPERIMENTAL, since = "5.5")
+fun assertDoesNotThrow(message: String, executable: () -> Unit): Unit =
+    assertDoesNotThrow({ message }, executable)
+
+/**
+ * Example usage:
+ * ```kotlin
+ * assertDoesNotThrow({ "Should not throw an exception }) {
+ *     // Code block that is expected to not throw an exception
+ * }
+ * ```
+ * @see Assertions.assertDoesNotThrow
+ */
+@API(status = EXPERIMENTAL, since = "5.5")
+fun assertDoesNotThrow(message: () -> String, executable: () -> Unit): Unit =
+    Assertions.assertDoesNotThrow(Executable(executable), Supplier(message))
```

Comment:
@mkobit Can you change these to returning `R`?
Change the executable's type to `() -> R`. That way you can do:
```kotlin
val something = assertDoesNotThrow { method() }
```

Reply:
Good suggestion - didn't even know those variants existed

Reply:
Neither did I until today when I ran into the problem where the java variants are unusable from Kotlin because the overload is ambiguous.